internal/adapters/operation/operation_flow_redis.go

```diff
@@ -68,36 +68,67 @@ func (r *redisOperationFlow) InsertOperationID(ctx context.Context, schedulerNam
-	var opIDs []string
+	var opID string
+
 	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
-		opIDs, err = r.client.BLPop(ctx, 0, r.buildSchedulerPendingOperationsKey(schedulerName)).Result()
+		opID, err = r.client.LIndex(ctx, r.buildSchedulerAuxiliaryPendingOperationsKey(schedulerName), 0).Result()
+
 		return err
 	})
+
+	if err == nil {
+		return opID, nil
+	}
+
+	if err != nil && err != redis.Nil {
+		return "", errors.NewErrUnexpected("failed to fetch next operation ID from auxiliary queue").WithError(err)
+	}
+
+	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
+		opID, err = r.client.BLMove(ctx, r.buildSchedulerPendingOperationsKey(schedulerName), r.buildSchedulerAuxiliaryPendingOperationsKey(schedulerName), "LEFT", "RIGHT", 0).Result()
+
+		return err
+	})
+
 	if err != nil {
-		return "", errors.NewErrUnexpected("failed to fetch next operation ID").WithError(err)
+		return "", errors.NewErrUnexpected("failed to fetch next operation ID from main queue").WithError(err)
 	}
 
-	if len(opIDs) == 0 {
-		return "", errors.NewErrNotFound("scheduler \"%s\" has no operations", schedulerName)
+	return opID, nil
+}
+
+// RemoveOperation removes the given operation from the operation flow.
+func (r *redisOperationFlow) RemoveOperation(ctx context.Context, schedulerName, operationID string) (err error) {
+	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
+		_, err = r.client.LPop(ctx, r.buildSchedulerAuxiliaryPendingOperationsKey(schedulerName)).Result()
+
+		return err
+	})
+	if err != nil && err != redis.Nil {
+		return errors.NewErrUnexpected("failed to pop from auxiliary pending operations list").WithError(err)
 	}
 
-	// Once redis finds any operation ID, it returns an array using:
-	// - the first position to indicate the list name;
-	// - the second position with the found value.
-	opId = opIDs[1]
-	return opId, nil
+	return nil
 }
 
 func (r *redisOperationFlow) ListSchedulerPendingOperationIDs(ctx context.Context, schedulerName string) (operationsIDs []string, err error) {
+	var operationsIDsFromAux []string
 	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
 		operationsIDs, err = r.client.LRange(ctx, r.buildSchedulerPendingOperationsKey(schedulerName), 0, -1).Result()
 		return err
 	})
 	if err != nil {
 		return nil, errors.NewErrUnexpected("failed to list pending operations for \"%s\"", schedulerName).WithError(err)
 	}
+	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
+		operationsIDsFromAux, err = r.client.LRange(ctx, r.buildSchedulerAuxiliaryPendingOperationsKey(schedulerName), 0, -1).Result()
+		return err
+	})
+	if err != nil && err != redis.Nil {
+		return nil, errors.NewErrUnexpected("failed to list pending operations for \"%s\"", schedulerName).WithError(err)
+	}
 
-	return operationsIDs, nil
+	return append(operationsIDs, operationsIDsFromAux...), nil
```

Comment:
Shouldn't operationIDsFromAux be in the front of operationIDs? This return put operations from the auxiliary list at the end of the line... Is this what we want here?

Reply:
It doesn't really matter the order of the IDs, since the list operation endpoint sorts operations by the createdAt.
But it makes sense to include ids from the aux in the beginning here